cellmlmanip/model.py

```diff
@@ -679,17 +679,23 @@ def convert_variable(self, original_variable, units, direction):
                           equations will be adjusted
                           or DataDirectionFlow.OUTPUT; the variable to be changed is an output, equations
                           are unaffected apart from converting the actual output
+        :param raise_errors: determines whether the method will throw an error if one of the checks fails
+                          if raise_errors==False and checks fail, the original variable is returned unchanged
         :return: new variable with desired units, or original unchanged if conversion was not necessary
         :throws: AssertionError if the arguments are of incorrect type
                                 or the variable does not exist in the model
                 DimensionalityError if the unit conversion is impossible
         """
         # assertion errors will be thrown here if arguments are incorrect type
-        self._check_arguments_for_convert_variables(original_variable, units, direction)
+        checks_passed = self._check_arguments_for_convert_variables(original_variable, units, direction, raise_errors)
+
+        # no conversion possible (checks failed)
+        if not checks_passed:
+            return original_variable
 
         original_units = original_variable.units
         # no conversion necessary
-        if original_units == units:
+        if self.units.get_conversion_factor(units, from_unit=original_units) == 1.0:
```

Comment:
I think I've found a simpler way to do this:
```
>>> import pint
>>> ureg = pint.UnitRegistry()
>>> ureg.define('my_mM = 0.001 * molar')
>>> ureg.my_mM == ureg.mM
False
>>> 1 * ureg.my_mM == 1 * ureg.mM
True
```
So we can just do here:
```python
if 1 * original_units == 1 * units:  # Still compares equal if the unit names differ
```

Reply:
ok